Assets/MixedRealityToolkit.Providers/XRSDK/Controllers/GenericXRSDKController.cs

```diff
@@ -0,0 +1,306 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.Input
+{
+    [MixedRealityController(
+        SupportedControllerType.GenericUnity,
+        new[] { Handedness.Left, Handedness.Right },
+        flags: MixedRealityControllerConfigurationFlags.UseCustomInteractionMappings)]
+    public class GenericXRSDKController : BaseController
+    {
+        public GenericXRSDKController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <summary>
+        /// The current pose of this XR SDK controller.
+        /// </summary>
+        protected MixedRealityPose CurrentControllerPose = MixedRealityPose.ZeroIdentity;
+
+        /// <summary>
+        /// The previous pose of this XR SDK controller.
+        /// </summary>
+        protected MixedRealityPose LastControllerPose = MixedRealityPose.ZeroIdentity;
+
+        /// <summary>
+        /// The current position of this XR SDK controller.
+        /// </summary>
+        protected Vector3 CurrentControllerPosition = Vector3.zero;
+
+        /// <summary>
+        /// The current rotation of this XR SDK controller.
+        /// </summary>
+        protected Quaternion CurrentControllerRotation = Quaternion.identity;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
+        {
+            AssignControllerMappings(DefaultInteractions);
+        }
+
+        /// <summary>
+        /// Update the controller data from XR SDK.
+        /// </summary>
+        public virtual void UpdateController(InputDevice inputDevice)
+        {
+            if (!Enabled) { return; }
+
+            if (Interactions == null)
+            {
+                Debug.LogError($"No interaction configuration for {GetType().Name}");
+                Enabled = false;
+            }
+
+            var lastState = TrackingState;
+            LastControllerPose = CurrentControllerPose;
+
+            // Check for position and rotation.
+            IsPositionAvailable = inputDevice.TryGetFeatureValue(CommonUsages.devicePosition, out CurrentControllerPosition);
+            IsPositionApproximate = false;
```

Comment:
>IsPositionApproximate [](start = 12, length = 21)

I think this is unused

Reply:
It's a public property accessible outside the class. Needs to be set for consistency with our other controller classes.